apps/condo/domains/user/schema/SignInUserService.js

```diff
@@ -0,0 +1,233 @@
+    },
+}
+
+
+const REQUIRED_USER_FIELDS_BY_TYPE = {
+    [RESIDENT]: ['phone'],
+    [STAFF]: ['phone', 'name', 'password'],
+}
+
+const getUserPayload = (user, userData) => {
+    const payload = {}
+    for (const [key, value] of Object.entries(userData))  {
+        if (!value) continue
+        if (get(user, key)) continue
+        payload[key] = value
+    }
+    return payload
+}
+
+const checkRequiredUserFields = (requiredFields, userData) => {
+    if (!Array.isArray(requiredFields)) throw new Error('"requiredFields" should be an array')
+
+    const missingFields = []
+    for (const field of requiredFields) {
+        const value = get(userData, field, null)
+        if (!value && typeof value !== 'boolean' && typeof value !== 'number') missingFields.push(field)
+    }
+    return { missingRequiredFields: missingFields.length > 0, missingFields }
+}
+
+const SignInUserService = new GQLCustomSchema('SignInUserService', {
+    types: [
+        {
+            access: true,
+            type: 'input SignInUserUserDataInput { phone: String, name: String, email: String, password: String, meta: JSON }',
+        },
+        {
+            access: true,
+            type: 'input SignInUserInput { dv: Int!, sender: JSON!, confirmActionToken: String!, userType: UserTypeType!, userData: SignInUserUserDataInput }',
+        },
+        {
+            access: true,
+            type: 'type SignInUserOutput { user: User, token: String! }',
+        },
+    ],
+
+    mutations: [
+        {
+            access: access.canSignInUser,
+            schema: 'signInUser(data: SignInUserInput!): SignInUserOutput',
+            doc: {
+                summary: 'This mutation authorizes the user without a password, after confirming the phone number.\n\n' +
+                    'If the user is not registered, then he will be created with the data that is passed in the payload (user data).\n\n' +
+                    'If the existing user is missing some fields, then these fields will be taken from the payload (user data) and updated.\n\n' +
+                    'This mutation is not available for service users!',
+                errors: Object.fromEntries(Object.entries(ERRORS).filter(([key, error]) => typeof error === 'object')),
+            },
+            resolver: async (parent, args, context, info, extra = {}) => {
+                const { data } = args
+                const { confirmActionToken, userType, userData, dv, sender } = data
+
+                const locale = extractReqLocale(context.req) || conf.DEFAULT_LOCALE
+
+                const dvAndSender = { dv, sender }
+                const userInfo = {
+                    ...userData,
+                    type: userType,
+                    name: String(get(userData, 'name', '')).trim() || null,
+                    email: normalizeEmail(get(userData, 'email')) || null,
+                    isEmailVerified: false,
+                    phone: normalizePhone(get(userData, 'phone')) || null,
```

Comment:
Isn't `normailzePhone` / `normailzeEmail` returning `null` if input is not provided